thrust/thrust/pair.h

```diff
@@ -73,10 +73,49 @@ using tuple_size = ::cuda::std::tuple_size<T>;
  *          provided by <tt>pair::second_type</tt>.
  */
 template <class T, class U>
-using pair = ::cuda::std::pair<T, U>;
-
-using ::cuda::std::get;
-using ::cuda::std::make_pair;
+struct pair : public _CUDA_VSTD::pair<T, U>
+{
+  using super_t = _CUDA_VSTD::pair<T, U>;
+  using super_t::super_t;
+
+#if (defined(_CCCL_COMPILER_GCC) && __GNUC__ < 9) || (defined(_CCCL_COMPILER_CLANG) && __clang_major__ < 12)
+  // For whatever reason nvcc complains about that constructor being used before being defined in a constexpr variable
+  constexpr pair() = default;
+
+  template <class _U1          = T,
+            class _U2          = U,
+            class _Constraints = typename _CUDA_VSTD::__pair_constraints<T, U>::template __constructible<_U1, _U2>,
+            _CUDA_VSTD::__enable_if_t<_Constraints::__implicit_constructible, int> = 0>
+  _CCCL_HOST_DEVICE constexpr pair(_U1&& __u1, _U2&& __u2)
+      : super_t(_CUDA_VSTD::forward<_U1>(__u1), _CUDA_VSTD::forward<_U2>(__u2))
+  {}
+#endif // _CCCL_COMPILER_GCC < 9 || _CCCL_COMPILER_CLANG < 12
+};
+
+#if _CCCL_STD_VER >= 2017
+template <class _T1, class _T2>
+_CCCL_HOST_DEVICE pair(_T1, _T2) -> pair<_T1, _T2>;
+#endif // _CCCL_STD_VER >= 2017
+
+template <class T1, class T2>
+inline _CCCL_HOST_DEVICE
+  _CUDA_VSTD::__enable_if_t<_CUDA_VSTD::__is_swappable<T1>::value && _CUDA_VSTD::__is_swappable<T2>::value, void>
+  swap(pair<T1, T2>& lhs, pair<T1, T2>& rhs) noexcept(
+    (_CUDA_VSTD::__is_nothrow_swappable<T1>::value && _CUDA_VSTD::__is_nothrow_swappable<T2>::value))
+{
+  lhs.swap(rhs);
+}
```

Comment:
Question: why isn't `swap` a hidden friend?

Reply:
Ah, I get it. Because:
```c++
template <typename T1, typename T2>
struct pair {
  friend _CUDA_VSTD::__enable_if_t<_CUDA_VSTD::__is_swappable<T1>::value && 
            _CUDA_VSTD::__is_swappable<T2>::value, void>
   swap(pair<T1, T2>& lhs, pair<T1, T2>& rhs) noexcept { ... }
};
```
would immediately fail for a `pair<nonswappable, int>`. You want to delay that failure to whether `swap` is actually called. So you would need:
```c++
template <typename T1, typename T2>
struct pair {
  template <typename U1 = T1, typename U2 = T2>
  friend _CUDA_VSTD::__enable_if_t<_CUDA_VSTD::__is_swappable<U1>::value && 
            _CUDA_VSTD::__is_swappable<U2>::value, void> ...
};
```
And I agree, this may be worse than where we started. Thx!